spring-5-reactive-client/src/test/java/com/baeldung/reactive/service/EmployeeServiceIntegrationTest.java

```diff
@@ -0,0 +1,121 @@
+package com.baeldung.reactive.service;
+
+import com.baeldung.reactive.model.Employee;
+import com.baeldung.reactive.enums.Role;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.Rule;
+import org.junit.jupiter.api.*;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class EmployeeServiceIntegrationTest {
+
+    public static MockWebServer mockBackEnd;
+    private EmployeeService employeeService;
+    private ObjectMapper MAPPER = new ObjectMapper();;
+
+    @BeforeAll
+    static void setUp() throws IOException {
+        mockBackEnd = new MockWebServer();
+        mockBackEnd.start();
+    }
+
+    @AfterAll
+    static void tearDown() throws IOException {
+        mockBackEnd.shutdown();
+    }
+
+    @BeforeEach
+    void initialize() {
+
+        String baseUrl = String.format("http://localhost:%s", mockBackEnd.getPort());
+        employeeService = new EmployeeService(baseUrl);
+    }
+
+    @Test
+    void getEmployeeById() throws Exception {
+
+        Employee mockEmployee = new Employee(100, "Adam", "Sandler", 32, Role.LEAD_ENGINEER);
+        mockBackEnd.enqueue(new MockResponse().setBody(MAPPER.writeValueAsString(mockEmployee))
+                .addHeader("Content-Type", "application/json"));
+
+        Mono<Employee> employeeMono = employeeService.getEmployeeById(100);
+
+        StepVerifier.create(employeeMono)
+                .expectNextMatches(employee -> employee.getRole().equals(Role.LEAD_ENGINEER))
+                .verifyComplete();
+
+        RecordedRequest recordedRequest = mockBackEnd.takeRequest();
+        assertEquals(recordedRequest.getMethod(), "GET");
+        assertEquals((recordedRequest.getPath()), "/employee/100");
```

Comment:
The expected and actual are reversed in both of these examples. There's also some extra `()` that aren't needed.